### PR TITLE
Move extra buildings submit login down to ExtraBuildingsField

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- `ExtraBuildingsField` now implement its own submit logic
- Split `useHandleSubmitPriceCalculator` into reusable `useUpdatePriceIntent` and specialized `useHandleSubmitPriceCalculatorSection`

## Justify why they are needed

- Allows us to remove single-use `onSubmit` abstraction from `AutomaticField`
- Less prop-drilling

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4424?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4424** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4424?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4423** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4423?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4419** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4419?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4420** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4420?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by guilhermespopolin — APPROVED** [review]
Nice one :)

File: apps/store/src/components/PriceCalculator/AutomaticField.tsx [hunk 1]
```diff
@@ -17,12 +16,11 @@ import { useTranslateFieldLabel } from './useTranslateFieldLabel'
 
 type Props = {
   field: InputFieldType
-  onSubmit: (data: JSONData) => Promise<unknown>
```
Comment (gustaveen) [inline apps/store/src/components/PriceCalculator/AutomaticField.tsx:20]: Nice! 👌

**Review by gustaveen — APPROVED** [review]
Great refactoring 🙌

File: apps/store/src/components/PriceCalculator/useUpdatePriceIntent.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    awaitRefetchQueries: true,
+    onCompleted: (data) => {
+      const { priceIntent } = data.priceIntentDataUpdate
+      const { shopSession: updatedShopSession } = data.shopSessionCustomerUpdate
+      if (updatedShopSession && priceIntent) {
+        onSuccess({ priceIntent, customer: updatedShopSession.customer ?? undefined })
+      }
+    },
+    onError(error) {
+      window.console.log("Couldn't update price intent", error)
```
[the remark this answers is not in the document]
Comment (alebedev) [reply apps/store/src/components/PriceCalculator/useUpdatePriceIntent.ts:36]: Fixed in later PR to reduce number of restacks

**Comment by alebedev** [comment]
### Merge activity

* **Jul 1, 8:02 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4424) rebased this pull request as part of a merge.

**Pull request merged**